tools/pkg/commands/list.toit

```diff
@@ -0,0 +1,98 @@
+// Copyright (C) 2024 Toitware ApS.
+// Use of this source code is governed by a Zero-Clause BSD license that can
+// be found in the tests/LICENSE file.
+
+import system
+
+import cli
+import encoding.json
+import encoding.yaml
+
+import ..pkg
+import ..registry
+import ..error
+import ..registry.description
+
+class ListCommand:
+  name/string?
+  verbose/bool
+  output/string
+
+  constructor parsed/cli.Parsed:
+    name = parsed[NAME-OPTION]
+    verbose = parsed[VERBOSE-OPTION]
+    output = parsed[OUTPUT-OPTION]
+
+  execute:
+    registry-packages := registries.list-packages
+    if name:
+      if not registry-packages.contains name:
+        error "Registry not found: $name"
+      registry-packages.filter --in-place: | k v | k == name
+
+    if output == "list":
+      registry-packages.do: | registry-name registry/Map |
+        print "$registry-name: $(registry["registry"].stringify)"
+        list-textual registry["descriptions"] --verbose=verbose --indent="  "
+    else:
+      result/Map := ?
+      if verbose:
+        result = registry-packages.map: | registry-name registry/Map |
+          { "registry": registry["registry"].to-map,
+            "packages": (registry["descriptions"].map: verbose-description it)
+          }
+      else:
+        result = registry-packages.map: | registry-name registry/Map |
+          { "registry": registry["registry"].stringify,
+            "packages": (registry["descriptions"].map: | description/Description | {
+              description.name : description.version
+            })
+          }
+      if output == "json":
+        print (json.stringify result)
+      else if output == "yaml":
+        print (yaml.stringify result)
+
+  static verbose-description description/Description --allow-extra-fields=false -> Map:
+    filtered := description.content.filter: | k _ |
+                  k != Description.NAME-KEY_ and
```

Comment:
I don't understand this block.
What is this function supposed to do?

I guess it removes the 'name' key, because that one is prefixed. (add a comment).
It then allows every field, but shouldn't the dependency-key and env-key checks be "==" and "or"?